libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -78,14 +78,11 @@ void UARTDriver::_begin(uint32_t b, uint16_t rxS, uint16_t txS)
 
     while (_in_timer) hal.scheduler->delay(1);
 
-    _device->set_speed(b);
-
     bool clear_buffers = false;
-    if (b != 0) {
-        if (_baudrate != b && hal.console != this) {
-            clear_buffers = true;
-        }
+    if (b != 0 && _baudrate != b && hal.console != this) {
```

Comment:
hal.console could be a real serial device and often is one on linux, so we do need to call set_speed() when b != 0 on the console

Reply:
So this should be something like the following?
```cpp
    if (b != 0 && _baudrate != b) {
        _device->set_speed(b);
        _baudrate = b;
        if (hal.console != this) {
            clear_buffers = true;
        }
    }
```